src/pages/Environments.js

```diff
@@ -21,15 +19,17 @@ import { reportErrorAndRethrow, withErrorHandling, withErrorIgnoring, withErrorR
 import Events from 'src/libs/events'
 import * as Nav from 'src/libs/nav'
 import { useCancellation, useGetter, useOnMount, usePollingEffect } from 'src/libs/react-utils'
-import {
-  defaultComputeZone, getAppCost, getComputeStatusForDisplay, getCurrentRuntime, getDiskAppType, getGalaxyComputeCost, getPersistentDiskCostMonthly,
-  getRegionFromZone, getRuntimeCost, isApp, isComputePausable, isGcpContext, isResourceDeletable, mapToPdTypes, workspaceHasMultipleApps,
-  workspaceHasMultipleDisks
-} from 'src/libs/runtime-utils'
 import { contactUsActive } from 'src/libs/state'
 import * as Style from 'src/libs/style'
 import { topBarHeight } from 'src/libs/style'
 import * as Utils from 'src/libs/utils'
+import { getToolFromRuntime, isPauseSupported, tools } from 'src/pages/workspaces/workspace/analysis/notebook-utils'
+import { SaveFilesHelp, SaveFilesHelpGalaxy } from 'src/pages/workspaces/workspace/analysis/runtime-common'
+import {
+  defaultComputeZone, getAppCost, getComputeStatusForDisplay, getCurrentRuntime, getDiskAppType, getGalaxyComputeCost, getPersistentDiskCostMonthly,
+  getRegionFromZone, getRuntimeCost, isApp, isComputePausable, isGcpContext, isResourceDeletable, mapToPdTypes, workspaceHasMultipleApps,
+  workspaceHasMultipleDisks
+} from 'src/pages/workspaces/workspace/analysis/runtime-utils'
```

Comment:
I think the fact that we're importing these modules from `pages/Environments` suggests that `pages//workspaces/workspace/analysis` is not the most appropriate place for them. I think they belong at a higher level.

Reply:
Yea, I agree that pages is not an appropriate place for these files. I want to get them in the same place first. Honestly, I think the top-level pages/components/libs directory structure itself is a bit outdated given the sprawl. I'm a bit hesitant to fix this though, as in my head it would require making `workspaces` the top-level folder, which would touch far more teams files than our own.

For now, I like getting all of IA files within one folder, and deferring the handling of the top-level structure to something more thought out. cc @msilva-broad , I feel this is something that should be done sooner rather than later though

Reply:
agreed that we want to be conservative for now, since a broader restructure will require some thought and buy-in.  This particular file might be fine in it's new location if the only consumer/s is the visual components.  As we later iterate on improving separation-of-concerns, we can revisit as needed.

Reply:
tho if all these runtime-utils exports are non-visual (data-crunchy) things, maybe they should live alongside the new ajax sub-folder next to their newly split out ajax sub-file.... (runtime.js)